### DISCUSSION TITLE
Popularity calculation for packages: Did it change or is there a bug?

### DISCUSSION THREAD

**Discussion by nohli** [dart-lang/pub-dev#7123] [General]
I recently uploaded a new [package](https://pub.dev/packages/firebase_testlab_detector) that I use in my own app, both during local debugging and Codemagic production builds.

With my other packages, their popularity typically rose above 0% within the first day.

However, for this recent package, there's been no change for four days. Additionally, the scores of my other packages, which usually fluctuated only slightly, haven't changed at all.

I happened to notice this and wasn't sure whether to open an issue.

**Comment by isoos** [comment]
Thanks for reporting this, it is a bug. The last timestamp of the popularity metrics dump seems to be `2023-10-08T18:13:16.000Z`. 
/cc @kevmoo @jonasfj

**Reply by nohli** [reply]
Is the timestamp for the popularity metrics dump publicly available?
From my observation, it seemed like it was updated once after your response but hasn't changed since then.

**Reply by jonasfj** [reply]
> Is the timestamp for the popularity metrics dump publicly available?

Not really, but the source code for pub.dev is public :rofl: 

And if you read all if it, you might stumble on a end-point like:
https://pub.dev/debug

Maybe, we should have hidden it better :see_no_evil:, though to be clear, there is nothing secret about `/debug` -- it's just not intended for users 

Be sure that this not a stable end-point, we'll definitely break it without caring if anyone is reading it. Some of it might also be out-of-date or just wrong, or hard to understand.

**Reply by nohli** [reply]
Ok thanks haha

It says
```
  "popularity": {
    "fetched": "2023-10-23T19:24:44.492271",
    "count": 27773,
    "dateRange": "2023-08-17T00:00:00.000Z - 2023-10-15T16:48:01.000Z"
  }
```

Does the `dateRange` mean it's stale again?

**Reply by nohli** [reply]
@isoos it looks like the popularity score hasn't changed for some time again.
